File: development/NET/samples/server/Console/CS/MyDaAddressSpaceRoot.cs
using Softing.OPCToolbox;
using Softing.OPCToolbox.Server;

namespace Console
{
	public class MyDaAddressSpaceRoot : DaAddressSpaceRoot
	{
		#region Public Methods
		//--------------------

		public override int QueryAddressSpaceElementData(string anElementId, out AddressSpaceElement anElement)
		{

			MyDaAddressSpaceElement element = new MyDaAddressSpaceElement();
			element.ItemId = anElementId;
			anElement = element;

			return (int)EnumResultCode.S_OK;

		}   //	end QueryAddressSpaceElementData

		//--
		#endregion

	}   //	end MyDaAddressSpaceRoot
}   //	end namespace

File: development/NET/samples/server/Console/VB/MyDaAddressSpaceRoot.vb
Imports Softing.OPCToolbox
Imports Softing.OPCToolbox.Server

Namespace Console
	Public Class MyDaAddressSpaceRoot
		Inherits DaAddressSpaceRoot
#Region "Public Methods"
		'--------------------

		Public Overloads Overrides Function QueryAddressSpaceElementData(ByVal anElementId As String, ByRef anElement As AddressSpaceElement) As Integer

			Dim element As New MyDaAddressSpaceElement
			element.ItemId = anElementId
			anElement = element

			Return EnumResultCode.S_OK

		End Function

		'--
#End Region

	End Class
End Namespace

